pds/contracts/PackNFT.cdc

```diff
@@ -88,168 +110,303 @@ pub contract PackNFT: NonFungibleToken, IPackNFT {
+
+            // Initial status is Sealed.
+            self.status = Status.Sealed
+
+            // Salt is nil until reveal.
             self.salt = nil
         }
     }
 
-    pub resource NFT: NonFungibleToken.INFT, IPackNFT.IPackNFTToken, IPackNFT.IPackNFTOwnerOperator {
-        pub let id: UInt64
-        pub let hash: [UInt8]
-        pub let issuer: Address
+    /// Resource that defines a Pack NFT.
+    ///
+    access(all) resource NFT: NonFungibleToken.NFT, IPackNFT.NFT, IPackNFT.IPackNFTToken, IPackNFT.IPackNFTOwnerOperator {
+        /// This NFT's unique ID.
+        ///
+        access(all) let id: UInt64
+
+        /// This NFT's commit hash, used to verify the IDs of the NFTs in the Pack.
+        ///
+        access(all) let hash: [UInt8]
 
-        pub fun reveal(openRequest: Bool){
+        /// This NFT's issuer.
+        access(all) let issuer: Address
+
+        /// Reveal a Sealed Pack resource.
+        ///
+        access(NonFungibleToken.Update | NonFungibleToken.Owner) fun reveal(openRequest: Bool) {
             PackNFT.revealRequest(id: self.id, openRequest: openRequest)
         }
 
-        pub fun open(){
+        /// Open a Revealed Pack resource.
+        ///
+        access(NonFungibleToken.Update |NonFungibleToken.Owner) fun open() {
             PackNFT.openRequest(id: self.id)
         }
 
-        destroy() {
-            let p <- PackNFT.packs.remove(key: self.id) ?? panic("no such pack")
-            PackNFT.totalSupply = PackNFT.totalSupply - (1 as UInt64)
+        /// Event emitted when a NFT is destroyed (replaces Burned event before Cadence 1.0 update)
+        ///
+        access(all) event ResourceDestroyed(id: UInt64 = self.id)
 
-            emit Burned(id: self.id)
-            destroy p
+        /// Executed by calling the Burner contract's burn method (i.e., conforms to the Burnable interface)
+        ///
+        access(contract) fun burnCallback() {
+            PackNFT.totalSupply = PackNFT.totalSupply - 1
+            destroy <- PackNFT.packs.remove(key: self.id) ?? panic("no such pack")
         }
 
-        init(commitHash: String, issuer: Address ) {
+        /// NFT resource initializer.
+        ///
+        view init(commitHash: String, issuer: Address) {
             self.id = self.uuid
             self.hash = commitHash.decodeHex()
             self.issuer = issuer
         }
 
+        /// Create an empty Collection for Pinnacle NFTs and return it to the caller
+        ///
+        access(all) fun createEmptyCollection(): @{NonFungibleToken.Collection} {
+            return <- PackNFT.createEmptyCollection(nftType: Type<@NFT>())
+        }
+
+        /// Return the metadata view types available for this NFT.
+        ///
+        access(all) view fun getViews(): [Type] {
+            return []
+        }
+
+        /// Resolve this NFT's metadata views.
+        ///
+        access(all) view fun resolveView(_ view: Type): AnyStruct? {
+            return nil
+        }
```

Comment:
Are you sure that this can't be represented by any views? I would think that at the minimum, you would at least want the `NFTCollectionData` and `NFTCollectionDisplay` views, but there might be even more that apply

Reply:
The views depend on each individual PackNFT instance. The AllDay version of the PackNFT contract updated in this PR defines the views. Added a comment in the vanilla PackNFT contract to make it clear that the views should be implemented 👍